### PR TITLE
Feat: Applications Page for Non-Profits

### PR DESCRIPTION
**Pull request by blst15ki**
## Applications Page
Added applications page for non-profits to view pending applications
image

Added application page on viewing an application
image

Added check to prevent non-profits from submitting multiple applications to the same grant
image

### CODE REVIEW

File: components/dashboard/ApplicationCard.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import Link from "next/link"
+import ChevronRightIcon from '@mui/icons-material/ChevronRight';
+import { formatDate } from "@/utils/helperFunctions";
+
+export default function ApplicationCard({appID, name, amount, description, createdDate}: {appID: number, name: string, amount: number, description: string, createdDate: string}){
```
Comment (nam-t24) [inline components/dashboard/ApplicationCard.tsx:5]: appCards

Could you move the button down to the bottom like the cards in open grants. The grant title and View Application are too long to fit in the same line. I think this would also help fit 1 more cards in the row, if not decrease card width by a hair so we could have 3 cards in a row.

File: utils/scripts/applications.ts [hunk 2]
```diff
@@ -122,4 +127,36 @@ export const getAcceptedApplicationByGrantID = async (grantID: number) => {
         return null;
     }
     return data;
+}
+
+export const getApplicationsByOrgID = async (orgID: number) => {
+    const supabase = createClient();
+
+    const { data, error } = await supabase.from('applications').select('id, created_at, FK_grantID(name, amount, description)').eq('FK_orgApply', orgID).eq('status', 'Pending');
```
Comment (nam-t24) [inline utils/scripts/applications.ts]: To get the 'in review' applications, you'll have to do a join query and get applications that are tied to grants that have inReview == true. 

Donor orgs can mark applications as accepted or rejected during their review so applications marked as accepted/rejected are still technically being reviewed if donor org did not finish application review and should still show up in the applications list. idt I explained that really well. 

Basically if i'm reviewing apps and mark an application as rejected they are not officially rejected since the grant is still in review and ppl in my org can unrejected/accept them. So that application should still show up for non-profit app list despite being labeled as rejected

File: app/dashboard/organization/grants/apply/[id]/page.tsx [hunk 3]
```diff
@@ -38,7 +38,16 @@ export default function Apply({ params }: { params: { id: string } }) {
         setSubmitting(true);
 
         const error = await createApplication(description, purpose, timeline, +params.id);
-        if (error) {
+        if(error === 'application exists') {
```
Comment (nam-t24) [inline app/dashboard/organization/grants/apply/[id]/page.tsx]: It would be better if we don't even allow the user on this page if they already applied. We don't want them to fill out a whole application only to submit and say they already submitted. But for the extent of this project I don't think it matters to change it

File: components/dashboard/ApplicationCard.tsx (hunk 1 above)
Comment (blst15ki) [reply components/dashboard/ApplicationCard.tsx:5]: Would it be better to move it down or shorten it to "View App"? My thought is that it would be good to keep the placement similar to the grant cards, but users wouldn't be able to see both anyway.

Comment (blst15ki) [reply components/dashboard/ApplicationCard.tsx:5]: Also, it's the same for grant cards in the organization grants page, should that be changed as well?

Comment (nam-t24) [reply components/dashboard/ApplicationCard.tsx:5]: We could do "View App". idk how grants cards are on your screen but if it's also only 2 cards in a row could you shorten them by a hair to fit 3

**Comment by blst15ki** [comment]
## Visual Changes
Application cards adjusted
image

Grant cards adjusted
image

**Comment by blst15ki** [comment]
Probably should add a feature to edit or delete an application like with grants but that can be addressed later

**Review by nam-t24 — APPROVED** [review]
